independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/utils/NestedMaps.java

```diff
@@ -46,10 +46,22 @@ public static void deepMerge(Map left, Map right) {
                 c.addAll((Collection) leftValue);
                 c.addAll((Collection) rightValue);
                 left.put(key, c);
-            } else {
```

Comment:
@aloubyansky I found this nasty bug (from myself) while using a static codestart catalog from the tests 😱 the codestart catalog data was muted when generating a project (luckily in prod it's read at each generation).. This is not very optimized, at some point I will probably cache it. So better it be fixed :)

Reply:
FYI in this `deepMerge`, it's ok to mutate left, but not right so I added some safety..